### ISSUE TITLE
question about mxnet implementation of CycleGAN

### ISSUE THREAD

**Issue by WillSuen** [dmlc/mxnet#6412]
I am trying to implement CycleGAN (https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix) with mxnet. I need to realize two generator networks for style translation between A and B, and the cycle loss will be 
```
inputA -> GeneratorB -> fakeB -> GeneratorA -> A'
inputB -> GeneratorA -> fakeA -> GeneratorB -> B'
loss = L1loss(inputA, A') + L1loss(inputB, B')
```
I tried to use `MakeLoss` function, but I am lost with how to make the cycle symbol. Is there an easy way to implement this kind of cycle network in mxnet? 

Really appreciate any suggestions and advice.

**Comment by Godricly** [comment]
I made 4 symbols and trained them one by one.

**Comment by Ldpe2G** [comment]
@WillSuen you can refer to my implementation:
https://github.com/Ldpe2G/DeepLearningForFun/tree/master/Mxnet-Scala/CycleGAN

**Comment by Ldpe2G** [comment]
My implementation is the same with @Godricly , made 4 symbols(Generator A, Generator B, Dscriminator A, Descriminator B). and train them one by one, you can also refer to the torch implementation:
 https://github.com/junyanz/CycleGAN, My implementation was based on it.

**Comment by WillSuen** [comment]
Thanks a lot! I'll have a try.
